algorithms/kernel/dtrees/forest/classification/df_classification_predict_dense_default_batch_impl.i

```diff
@@ -729,6 +749,19 @@ Status PredictClassificationTask<algorithmFPType, cpu>::predictByBlocksOfTrees(s
         });
         s = safeStat.detach();
     }
+    if(probBDPtr != nullptr)
+    {
+        daal::threader_for(dim.nDataBlocks, dim.nDataBlocks, [&](size_t iBlock)
+        {
+            const size_t iStartRow = iBlock * dim.nRowsInBlock;
+            const size_t nRowsToProcess = (iBlock == dim.nDataBlocks - 1) ? dim.nRowsTotal - iStartRow : dim.nRowsInBlock;
+            algorithmFPType* prob = probBDPtr + iStartRow * _nClasses;
+            daal::threader_for(nRowsToProcess, nRowsToProcess, [&](size_t iRow)
```

Comment:
You create nRowsToProcess TBB tasks here, it is unefficient way.
Better to create some blocks of rows